workflows_core/engine/abstract_engine.py

```diff
@@ -102,26 +115,44 @@ def iterate(
         self,
         filters: Optional[List[Filter]] = None,
         select_fields: Optional[List[str]] = None,
+        max_retries: int = 3,
     ):
-        if filters is not None:
-            filters += self._get_workflow_filter()
-        else:
-            filters = self._get_workflow_filter()
-        while True:
-            chunk = self._dataset.get_documents(
-                self._chunksize,
-                filters=filters if filters is not None else self._filters,
-                select_fields=select_fields
-                if select_fields is not None
-                else self._select_fields,
-                after_id=self._after_id,
-                worker_number=self.worker_number,
-            )
-            self._after_id = chunk["after_id"]
-            if not chunk["documents"]:
-                break
-            yield chunk["documents"]
+        if filters is None:
+            filters = self._filters
 
-    def update_chunk(self, chunk: List[Document]):
+        if select_fields is None:
+            select_fields = self._select_fields
+
+        retry_count = 0
+        while True:
+            try:
+                chunk = self._dataset.get_documents(
+                    self._chunksize,
+                    filters=filters,
+                    select_fields=select_fields,
+                    after_id=self._after_id,
+                    worker_number=self.worker_number,
+                )
+            except ConnectionError as e:
+                logger.error(e)
+                retry_count += 1
+                if retry_count >= max_retries:
+                    raise MaxRetriesError("max number of retries exceeded")
```

Comment:
Don't re-try immediately. This won't solve the problem. Let it wait for 1 second first before re-trying.

Reply:
Make sure the 1 second is adjustable